features/steps/command/shell.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Running shell commands
+
+  You can run an *ad hoc* script with the following step
+  - `When I run the following script:`
+
+  Or you can run shell commands with:
+  - `I run the following (bash|zsh|fish|dash)? commands
+
+  Background:
+    Given I use a fixture named "cli-app"
+
+  Scenario: Running script
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running scripts
+      Scenario: Running ruby script
+        When I run the following script:
```

Comment:
I know it's a little bit annoying for you :sweat_smile:, but what do you think
about the following idea: Let's merge the new steps you want to add into a
single one.

~~~ruby
When(/^I run the following commands(?: with `([^`]+))?`:$/) do |shell, commands|
  prepend_environment_variable('PATH', expand_path('bin') + ':')

  shell ||= Aruba.platforms.default_shell # => should return 'bash' on unix

  # Be care full I change generator to interpreter here
  ScriptFile.new(:interpreter => shell, :content => file_content,
                 :path => expand_path('bin/myscript')).call
  step 'I run `myscript`'
end
~~~

If we support partial strings, users can pass `/usr/bin/env ruby` to the step
and the correct generator is found for this too. If the user passes
`/usr/bin/bash` to the step, this would be used instead of `/usr/bin/env bash`. Would
this work for your use case? If you like that idea and implement it, can you
also add cucumber examples for that?

~~~ruby
class AbstractScript
  private

  attr_reader :path, :content, :interpreter

  public

  def initialize(path, content, interpreter)
    @path        = path
    @content     = content
    @interpreter = interpreter
  end

  def self.match?(matcher)
    matcher =~ Regexp.new(shell)
  end

  def call
    h = if interpreter_is_absolute_path
          format('#! %s', interpreter)
        elsif interpreter_is_just_the_name_of_shell
          header
        else
          format('#! %s', interpreter)
        end

    text = "#{h}\n#{content}"

    Aruba.platform.write_file(path, text)
    Aruba.platform.chmod('0755', path)
  end

  private

  def interpreter_is_absolute_path
    Aruba.platform.absolute_path? interpreter
  end

  def interpreter_is_just_the_name_of_shell?
    interpreter == shell
  end
end
~~~


What do you think about that? Using this code we could get rid of the
`self.name` method. Personally I like that idea. ;-)

~~~ruby
class AbstractScript
  def self.shell
    name.gsub(/^([^:]+::)/, '').gsub(/File$/, '').downcase
  end

  def shell
    self.shell
  end
end
~~~